contracts/contracts/vault/OETHVaultAdmin.sol

```diff
@@ -1,12 +1,112 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { IVault } from "../interfaces/IVault.sol";
 import { VaultAdmin } from "./VaultAdmin.sol";
 
 /**
  * @title OETH VaultAdmin Contract
  * @author Origin Protocol Inc
  */
 contract OETHVaultAdmin is VaultAdmin {
+    using SafeERC20 for IERC20;
+
+    address public immutable weth;
+
+    constructor(address _weth) {
+        weth = _weth;
+    }
+
+    /// @dev Simplified version of the deposit function as WETH is the only supported asset.
+    function _depositToStrategy(
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal override {
+        require(
+            strategies[_strategyToAddress].isSupported,
+            "Invalid to Strategy"
+        );
+        require(
+            _assets.length == 1 && _amounts.length == 1 && _assets[0] == weth,
+            "Only WETH is supported"
+        );
+
+        // Check the there is enough WETH to transfer once the WETH reserved for the withdrawal queue is accounted for
+        require(_amounts[0] <= _wethAvailable(), "Not enough WETH available");
```

Comment:
nit: Short-circuiting for `_amounts[0] == 0` would be nice

Reply:
_wethAvailable() will be > 0 for the deposit to work. It's basically all the WETH that is not reserved for the withdrawal queue.

Reply:
Not about `_wethAvailable` but I was just wondering if `_amounts[0]` will ever be called with `0`. We have faced this issue while removing a few strategies (where we call withdrawFromAllStrategies and it failed because we didn't have a zero-amount check and we had to resort to workarounds)

Reply:
I see, you want to revert the `depositToStrategy` if the asset amount is zero. It's not something we have previously done.

The issue with `withdrawFromAllStrategies` was some strategies didn't support withdrawing a zero amount. We had to add a tiny amount so we could do a withdrawal. Checking `depositToStrategy` amounts are not zero would not have prevented this.

Reply:
Not revert, but I meant to say just a `return` for zero deposits. 

> Checking depositToStrategy amounts are not zero would not have prevented this.

That's true